hera_qm/xrfi.py

```diff
@@ -1191,29 +1410,136 @@ def xrfi_run(ocalfits_file, acalfits_file, model_file, data_file, history,
     # Flag on combined metrics
     uvf_f2 = flag(uvf_metrics2, nsig_p=sig_init)
     uvf_fws2 = watershed_flag(uvf_metrics2, uvf_f2, nsig_p=sig_adj, inplace=False)
+    uvf_fws2.label = 'Flags from combined metrics, round 2.'
     uvf_combined2 = (uvf_fws2 | uvf_ogf2 | uvf_oxf2 | uvf_agf2 | uvf_axf2
-                     | uvf_vf2 | uvf_df2 | uvf_init)
+                     | uvf_vf2 | uvf_df2 | uvf_chisq_f2 | uvf_init)
+    uvf_combined2.label = 'ORd flags, round 2.'
+
+    # Write everything out
+    uvf_dict = {'apriori_flags.h5': uvf_apriori,
+                'v_metrics1.h5': uvf_v, 'v_flags1.h5': uvf_vf,
+                'og_metrics1.h5': uvf_og, 'og_flags1.h5': uvf_ogf,
+                'ox_metrics1.h5': uvf_ox, 'ox_flags1.h5': uvf_oxf,
+                'ag_metrics1.h5': uvf_ag, 'ag_flags1.h5': uvf_agf,
+                'ax_metrics1.h5': uvf_ax, 'ax_flags1.h5': uvf_axf,
+                'chi_sq_renormed1.h5': uvf_chisq, 'chi_sq_flags1.h5': uvf_chisq_f,
+                'combined_metrics1.h5': uvf_metrics, 'combined_flags1.h5': uvf_fws,
+                'flags1.h5': uvf_init,
+                'v_metrics2.h5': uvf_v2, 'v_flags2.h5': uvf_vf2,
+                'og_metrics2.h5': uvf_og2, 'og_flags2.h5': uvf_ogf2,
+                'ox_metrics2.h5': uvf_ox2, 'ox_flags2.h5': uvf_oxf2,
+                'ag_metrics2.h5': uvf_ag2, 'ag_flags2.h5': uvf_agf2,
+                'ax_metrics2.h5': uvf_ax2, 'ax_flags2.h5': uvf_axf2,
+                'data_metrics2.h5': uvf_d2, 'data_flags2.h5': uvf_df2,
+                'chi_sq_renormed2.h5': uvf_chisq2, 'chi_sq_flags2.h5': uvf_chisq_f2,
+                'combined_metrics2.h5': uvf_metrics2, 'combined_flags2.h5': uvf_fws2,
+                'flags2.h5': uvf_combined2}
 
-    # Threshold
-    uvf_temp = uvf_combined2.copy()
-    uvf_temp.to_metric(convert_wgts=True)
-    uvf_final = flag(uvf_temp, nsig_p=1.0, nsig_f=freq_threshold, nsig_t=time_threshold)
+    basename = qm_utils.strip_extension(os.path.basename(data_file))
+    for ext, uvf in uvf_dict.items():
+        outfile = '.'.join([basename, ext])
+        outpath = os.path.join(dirname, outfile)
+        uvf.write(outpath, clobber=clobber)
 
-    # Write out final metrics and flags
-    outfile = '.'.join([basename, final_metrics_ext])
-    outpath = os.path.join(dirname, outfile)
-    uvf_metrics2.write(outpath, clobber=clobber)
-    outfile = '.'.join([basename, final_flags_ext])
-    outpath = os.path.join(dirname, outfile)
-    uvf_final.write(outpath, clobber=clobber)
 
-    # Save calfits with new flags
```

Comment:
Is this functionality included in a different function?

Reply:
Yes, this is now done in the `day_threshold_run` wrapper.